src/bucket/BucketManagerImpl.cpp

```diff
@@ -350,14 +363,10 @@ BucketManagerImpl::assumeState(HistoryArchiveState const& has)
         mBucketList.getLevel(i).setSnap(snap);
         mBucketList.getLevel(i).setNext(has.currentBuckets.at(i).next);
     }
+
+    // we will need these buckets after restart
+    retainAll(has);
```

Comment:
this should no-op... The loop right before is supposed to do just that:
```
mBucketList.getLevel(i).setCurr(curr);
mBucketList.getLevel(i).setSnap(snap);
```
where `setCurr` and `setSnap` are calling `setRetain(true)` on basically all buckets in `has`

Reply:
oh I see there are the future buckets in there as well

Reply:
mmm, it looks like it's the same change than what you did in `LedgerManagerImpl::storeCurrentLedger()` (so logic should be shared).

I think this change will fix this specific issue, as GC will take care of things, but it seems we're chasing a larger underlying problem and playing wack-a-mole.

Is it maybe that `restartMerges` only marks `retain(true)` for buckets involved in new merges but doesn't actually do anything if a merge was already completed?